### PR TITLE
[SECURITY-904] Fold tests back into normal files

### PR DESCRIPTION
**Pull request by jsoref**
See [SECURITY-904](https://issues.jenkins-ci.org/browse/SECURITY-904).

I'm assuming enough time has elapsed that the TODO can be addressed

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@Wadeck

### CODE REVIEW

File: core/src/test/java/hudson/FilePathTest.java [hunk 1]
```diff
@@ -64,12 +65,8 @@
...
 import org.apache.tools.ant.taskdefs.Chmod;
 import static org.hamcrest.CoreMatchers.is;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertThat;
```
Comment (Wadeck) [inline core/src/test/java/hudson/FilePathTest.java]: nit: That one could be also included in the `.*`

File: test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java [hunk 2]
```diff
@@ -74,6 +76,15 @@
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.StaplerResponse;
 
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.not;
```
Comment (Wadeck) [inline test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java]: nit: Same here

**Review by Wadeck — APPROVED** [review]
Thank you for the task! 👍 

Just quick note, some of the star imports already contain other, so that could be simplify a bit.

File: test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java (hunk 2 above)
Comment (jsoref) [reply test/src/test/java/hudson/model/DirectoryBrowserSupportTest.java]: This can't be folded because a `*` is already contributing an `equalTo`

**Comment by jsoref** [comment]
@Wadeck: I can't figure out what file this is targeting: https://github.com/jenkinsci/jenkins/blob/584c34cda584b98b71d82d038e7d5ebe38534e18/test/src/test/java/hudson/security/HudsonPrivateSecurityRealm2SEC1158Test.java#L39
In fact, there are a bunch of files in that directory which seem like they might want to be folded, but they don't have comments.

n.b. That will be a distinct PR as I don't have enough info and they're distinct bugs.

**Comment by Wadeck** [comment]
@jsoref Due to [caccd02](https://github.com/jenkinsci/jenkins/commit/caccd027fae0778ab5e5133f57b7d7698a385045#diff-70cb8fa9169218ee4752db08f251ae5c) in jenkins-2.80, the `HudsonPrivateSecurityRealmTest` and `HudsonPrivateSecurityRealm2Test` were merged together. At the time of SEC-1158, I created it against jenkins-2.73, where both files existed. Long story short, it could be merged to the resulting `HudsonPrivateSecurityRealmTest`

**Comment by jsoref** [comment]
What I meant was that distinct bugs added those test files and cleaning up from them could be reasonably treated distinctly.

File: core/src/test/java/hudson/UtilTest.java [hunk 3]
```diff
@@ -545,4 +540,41 @@ public void testDifferenceDays() throws Exception {
     private Date parseDate(String dateString) throws ParseException {
         return new SimpleDateFormat("yyyy-MM-dd hh:mm:ss").parse(dateString);
     }
+
+    @Test
```
Comment (daniel-beck) [inline core/src/test/java/hudson/UtilTest.java]:
```suggestion
    @Test
    @Issue("SECURITY-904")
```

WDYT @Wadeck ?

**Review by daniel-beck — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]

**Review by batmat — APPROVED** [review]

**Comment by batmat** [comment]
Given only core/src/test was touched, I think no changelog is fine.

**Pull request merged**